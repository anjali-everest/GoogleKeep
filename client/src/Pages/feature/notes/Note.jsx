import React, { useState } from 'react'
import { useDispatch } from 'react-redux'
import { updateIsEditingNote, updateEditingNoteId, updateDeletingNoteId } from "./notesSlice"
import { deleteOneNote } from "./ThunkHandler"
import deleteIcon from "../../../styles/deleteIcon.png"

export const Note = (props) => {
    const { note } = props
    const dispatch = useDispatch()

    const [showOptions, setShowOptions] = useState(false)

    const onMouseEntered = () => { setShowOptions(!showOptions) }
    const onMouseLeft = () => { setShowOptions(!showOptions) }
    const onNoteClicked = (e) => {
        dispatch(updateIsEditingNote())
        dispatch(updateEditingNoteId(e.target.id))
    }
    const onDeleteNoteClicked = (e) => {
        e.stopPropagation()
        dispatch(updateDeletingNoteId(e.target.id))
        dispatch(deleteOneNote(e.target.id))
    }

    return (
        note && note.title && note.content && <div id={note.id} key={note.id} className="note-body" onMouseEnter={onMouseEntered} onMouseLeave={onMouseLeft} onClick={onNoteClicked}>
            <p id={note.id} className="note-title">{note.title}</p>
            <p id={note.id} className="note-content">{note.content.substring(0, 100)}</p>
            <img hidden={!showOptions} src={deleteIcon} alt="deleteIcon" className="note-delete-icon" onClick={onDeleteNoteClicked} />
        </div>
    )
}