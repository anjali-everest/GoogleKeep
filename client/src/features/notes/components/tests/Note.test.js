import React from "react";
import { shallow } from "enzyme";
import { Note } from "../Note";
import configureMockStore from "redux-mock-store";
import * as ReactReduxHooks from "react-redux";

const mockStore = configureMockStore();

describe("Note tests", () => {
  const noteProp = { id: 1, title: "title", content: "content" };
  const store = mockStore({
    notes: [],
    status: "idle",
    error: null,
    editingNoteId: null,
    isEditingNote: false,
    deletingNoteId: null,
  });

  beforeEach(() => {
    jest
      .spyOn(ReactReduxHooks, "useSelector")
      .mockImplementation(() => store.getState());
    jest
      .spyOn(ReactReduxHooks, "useDispatch")
      .mockImplementation(() => store.dispatch);
  });

  it("Should render without crashing", () => {
    shallow(<Note note={noteProp} />);
  });

  it("should match the Note snapshot", function () {
    const note = shallow(<Note note={noteProp} />);
    expect(note).toMatchSnapshot();
  });
});
