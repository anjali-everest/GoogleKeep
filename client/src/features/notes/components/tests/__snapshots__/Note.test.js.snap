// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note tests should match the Note snapshot 1`] = `ShallowWrapper {}`;
